File: benchmark/src/ruby/string_sort.rb
N=20000;seed=12345;def ri(m);$seed=($seed*16807)%2147483647;$seed%m;end;$seed=12345
arr=N.times.map{"w#{ri(100000)}"}
def ms(a) return a if a.length<=1;m=a.length/2;l=ms(a[0...m]);r=ms(a[m..]);res=[];pi=qi=0
 while pi<l.length;break if qi>=r.length;res<<(l[pi]<=r[qi] ? l[pi] : r[qi]);l[pi]<=r[qi] ? pi+=1 : qi+=1 end
 res.concat(l[pi..]);res.concat(r[qi..]);res end
start=Time.now.to_f*1000;s=ms(arr);acc=0;N.times{|i|acc=(acc+s[i].length*(i%13))%1000003}
finish=Time.now.to_f*1000;puts"Result: #{acc}";puts"Time: #{(finish-start).round}ms"

File: benchmarks/src/nodejs/string_sort.js
const N=20000;let seed=12345;function ri(m){seed=(seed*16807)%2147483647;return seed%m}
let arr=[];for(let i=0;i<N;i++)arr.push('w'+ri(100000));
function ms(a){if(a.length<=1)return a;let m=Math.floor(a.length/2),L=ms(a.slice(0,m)),R=ms(a.slice(m)),res=[],pi=0,qi=0;
 while(pi<L.length){if(qi>=R.length)break;if(L[pi]<=R[qi])res.push(L[pi++]);else res.push(R[qi++])}
 while(pi<L.length)res.push(L[pi++]);while(qi<R.length)res.push(R[qi++]);return res}
const start=Date.now();let s=ms(arr),acc=0;for(let i=0;i<N;i++)acc=(acc+s[i].length*(i%13))%1000003;
const end=Date.now();console.log("Result: "+acc);console.log("Time: "+Math.floor(end-start)+"ms")
